File: packages/duit_js/src/attributes/text_attributes.ts
import { TextAlign } from "./text_align";
import { TextDirection } from "./text_direction";
import { TextOverflow } from "./text_overflow";
import { TextStyle } from "./text_style";

export interface TextAttributes {
    data: string;
    textAlign?: keyof typeof TextAlign;
    textDirection?: keyof typeof TextDirection;
    softWrap?: boolean;
    overflow?: keyof typeof TextOverflow;
    textScaleFactor?: number;
    maxLines?: number;
    semanticsLabel?: string;
    style?: TextStyle;
}

File: packages/duit_go/pkg/duit_attributes/text_attributes.go
package duit_attributes

type TextAttributes struct {
	Data            string        `json:"data"`
	SemanticsLabel  string        `json:"semanticsLabel,omitempty"`
	TextAlign       TextAlign     `json:"textAlign,omitempty"`
	TextDirection   TextDirection `json:"textDirection,omitempty"`
	TextOverflow    TextOverflow  `json:"textOverflow,omitempty"`
	SoftWrap        bool          `json:"softWrap,omitempty"`
	TextScaleFactor float32       `json:"textScaleFactor,omitempty"`
	MaxLines        int           `json:"maxLines,omitempty"`
}
